### PR TITLE
feat: allow optional response status code validation

### PR DESCRIPTION
**Pull request by sceccotti89**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please fix the linting?

**Comment by sceccotti89** [comment]
@mcollina Fixed.

**Comment by mcollina** [comment]
Can you please add a test for the type https://github.com/fastify/fastify-response-validation/blob/master/types/index.test-d.ts?

**Comment by sceccotti89** [comment]
@mcollina Thanks for the feedback. Test added.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by sceccotti89** [comment]
My last forced push commit was to solve the linting issue. It should all be fine now.

File: index.js [hunk 1]
```diff
@@ -35,15 +35,16 @@ function fastifyResponseValidation (fastify, opts, next) {
...
 
-  function onRoute (opts) {
-    if (opts.responseValidation === false) return
-    if (opts.schema && opts.schema.response) {
-      opts.preSerialization = opts.preSerialization || []
-      opts.preSerialization.push(buildHook(opts.schema.response))
+  function onRoute (routeOpts) {
+    if (routeOpts.responseValidation === false) return
+    if (routeOpts.schema && routeOpts.schema.response) {
+      const responseStatusCodeValidation = routeOpts.responseStatusCodeValidation || opts.responseStatusCodeValidation
```
Comment (Uzlopak) [inline index.js:41]:
```suggestion
      const responseStatusCodeValidation = routeOpts.responseStatusCodeValidation || opts.responseStatusCodeValidation || false
```

File: index.js [hunk 2]
```diff
@@ -67,6 +68,11 @@ function fastifyResponseValidation (fastify, opts, next) {
     function preSerialization (req, reply, payload, next) {
       let validate = statusCodes[reply.statusCode] || statusCodes[(reply.statusCode + '')[0] + 'xx']
 
+      if (validate === undefined && responseStatusCodeValidation) {
```
Comment (Uzlopak) [inline index.js:71]:
```suggestion
      if (responseStatusCodeValidation && validate === undefined) {
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I personally would have created another preSerialization Hook Handler for the case of responseStatusCodeValidation was true. But I would say, that by setting the responseStatusCodeValidation fallback to false and checking in the if clause for the responseStatusCodeValidation being true or false v8 optimizer has a chance to optimize it.

**Comment by sceccotti89** [comment]
@Uzlopak PR updated.

**Review by Uzlopak — APPROVED** [review]
LGTM

**Comment by sceccotti89** [comment]
Hi guys. Any chance this can get merged soon?

**Pull request merged**